Libraries/Image/RCTDownloadTaskWrapper.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+typedef void (^RCTDataCompletionBlock)(NSData *data, NSError *error);
+typedef void (^RCTDataProgressBlock)(int64_t written, int64_t total);
+
+@interface RCTDownloadTaskWrapper : NSObject <NSURLSessionDownloadDelegate>
+
+@property(nonatomic,assign) id<NSURLSessionDownloadDelegate> delegate;
+
+
+// completionBlock will contain the block you're going to call when the request is completed
+@property (copy, nonatomic) RCTDataCompletionBlock completionBlock; //void (^completionBlock)(RCTDataCompletionBlock *);
+
+// progressBlock will contain the block to trigger a downloadProgress event
+@property (copy, nonatomic) RCTDataProgressBlock progressBlock;//void (^progressBlock)(RCTDataProgressBlock);
+
+
+// Create download task
+- (NSURLSessionDownloadTask *) downloadData:(NSURL *) url
```

Comment:
I would recommend creating an `-init` method that takes in an `NSURLSession`, rather than creating a new one every time.